Initiate Morpheus Dark mode

tzi opened the pull request:
# Description

This PR introduces dark mode for the main authenticated Matomo UI. ( UX-120 )

## How to use it

  - Dark mode is available in the authenticated Morpheus-based UI.
  - Go to Personal > General Settings, use the Theme setting, choose `Light (default)` or `Dark`, then save.
  - The preference is stored per user, not globally for the whole Matomo instance.
  - The switch applies immediately in the current document by updating the root theme attribute after save.
  - Dark mode is not available during installation.
  - Dark mode is not available on login and other pre-authentication pages.
  - HTML report emails should stay in light mode for a better mailbox compatibility.

## No matching browser preferences for now

The codebase is already prepared for browser-preference matching, but this PR does not expose that option to users yet.

  - auto theme mode support exists in the theme infrastructure.
  - The current user preference only exposes Light and Dark.
  - We want more coverage before enabling browser matching as a user-facing option.
  - Today, generated graphs can stay in the previous mode when the browser/OS preference changes.
  - In practice, those graphs need a page refresh or widget refresh to be redrawn in the new mode.

## How it works

Theme values are now defined in PHP as either a single value or a [light, dark] pair. For example:

```php
  /* before */
  public $colorText = '#212121';
  /* after */
  public $colorText = ['#212121', '#ccc'];
```

Those values are converted into CSS custom properties and LESS variables at asset-build time.

  - `ThemeStyles::toLessCode()` generates `:root` CSS custom properties for the light values.
  - It also generates dark overrides of this CSS custom properties for `[data-theme-mode="dark"]`.
  - It also generates browser-matching dark overrides inside `@media (prefers-color-scheme: dark)` for `[data-theme-mode="auto"]`.
  - The same method creates LESS aliases such as @theme-color-text: ~"var(--theme-color-text)".

This means most LESS can simply use shared theme tokens such as:

```css
  color: @theme-color-text;
  background: @theme-color-background-contrast;
  border-color: @theme-color-border;
```

The active mode is selected through a root data attribute on the main layout:

  - data-theme-mode="light"
  - data-theme-mode="dark"
  - data-theme-mode="auto" (prepared, but not exposed yet)

The front end updates that attribute when the user saves the setting, so normal CSS-driven parts of the UI switch without a full reload.

We also added an `.inDarkMode({ ... })` mixin for exceptions. Example:

```css
.some-widget {
  .inDarkMode({
    filter: none;
    background: @theme-color-background-contrast;
  });
```

That mixin is useful for true mode-specific edge cases, but it should be the exception, not the default. The preferred approach is to share the same LESS rule in both modes by using theme variables, because that keeps the code smaller and reduces maintenance cost.

## Side effects (why this PR is so big)

To make dark mode possible, this PR also cleans up older styling and assets.

  - Many hard-coded colors were migrated to theme variables.
  - In some places the new light-mode color is very close to, but not exactly, the previous literal color.
  - As a result, this PR changes a large number of screenshots with very small visual diffs.
  - Some PNG assets had to be cleaned up to remove solid white backgrounds and make them transparent.
  - Some small PNG icons were migrated to SVG so they render more cleanly across themes.
  - ExampleTheme support dark mode too so we can make dark UI Screenshots

## Risks

  - ⚠️ **This PR changes how theme values are exposed:** theme LESS variables now resolve to CSS custom properties, so external theme plugins that apply LESS color functions such as `lighten()`, `darken()`, or `fade()` directly to theme variables will no longer compile and must be updated.
  - ⚠️ **ThemeStyles is now effectively a broader public API change:** many public properties changed from scalar strings to string|array, and the constructor now requires a mode argument. That is fine inside core, but it is a compatibility risk for
    extensions/themes that instantiate or inspect ThemeStyles directly.
  - Any remaining hard-coded color, image background, or legacy widget styling can stand out in dark mode.
  - Premium plugins and plugin submodules may still contain light-only styling and are not fully audited by this PR.
  - Contrast and accessibility still need continued validation on less-travelled admin pages and complex widgets.

## In scope
  
### Dashboard

**Data:**

  - Data tables
  - Table configuration (the controls used to customize a report table, such as columns, sorting, filters, and display options)
  - Limit selector (the selector that changes how many rows are displayed in a report or widget)
  - Sparklines
  - jqPlot-based graphs
  - Series picker
  - Single-metric visuals

**Core dashboard UI:**

  - CoreHome layout
  - Comparisons
  - Site selector
  - Segment Editor UI

**Specific widget and data visualization:**

  - Live / real-time UI
  - Visitor Map styling and related visual assets
  - Visitor profile related surfaces
  - Row evolution
  - Transitions popup/report styling

### Shared UI primitives

  - Alerts (Info, Warning, Error)
  - Forms
  - some jQuery UI 
  - Buttons
  - Cards
  - Panels
  - Navs
  - Popups
  - Expandable lists (Visible in the Segment Editor)
  - Autocomplete popups (Visible in the Segment Editor)

### All Websites

  - MultiSites / All Websites visuals
  - Small trend/status icons migrated away from PNG-only rendering
  - Sparklines

### Administration

**Plugin and settings administration:**

  - Plugin administration screens
  - Manage Plugins row states
  - Diagnostics config file page

**Marketplace:**

  - Marketplace plugin details styling

## Known as out of scope

  - Table headings looks weird
  - Installation pages, which still use their own dedicated installation layout and CSS.
  - Login, invitation, password-reset, and other pre-authentication pages.
  - A full premium-plugin audit, but we already know some glitches:
      - Administration > Settings > SAML > Access to SP metadata > The Entity ID color field
      - Administration > Websites > Session recording > Create new session recording > URL validator helper loader
      - Administration > Websites > Custom Reports > Create new report > Dimensions & Metrics dropdown
      - Dashboard > Visitors > Real-time Map > the timing is hard to read (inverted box-shadow)
      - Dashboard > Funnels > See a Funnel > Heading in Funnel Report
      - TagManager > Create new container > Context title is dark
      - TagManager > Deep pages > a lot of tables or block are white / whity
      - Administration > Websites > Custom Reports > See a report > An extra white line at the left bottom of table
      - Dashboard > Behaviour > Users Flow > label are dark
      - Dashboard > Visitors > Cohorts > Cohorts Table

## Plugins update (screenshots)

This PR will have impact on plugins UI screenshots:

- [AnonymousPiwikUsageMeasurement #102](https://github.com/matomo-org/plugin-AnonymousPiwikUsageMeasurement/pull/102)
- [Bandwidth #106](https://github.com/matomo-org/plugin-Bandwidth/pull/106)
- [CustomAlerts #239](https://github.com/matomo-org/plugin-CustomAlerts/pull/239)
- [MarketingCampaignsReporting #207](https://github.com/matomo-org/plugin-MarketingCampaignsReporting/pull/207)
- [QueuedTracking #316](https://github.com/matomo-org/plugin-QueuedTracking/pull/316)
- [TagManager #1086](https://github.com/matomo-org/tag-manager/pull/1086)
- [TrackingSpamPrevention #159](https://github.com/matomo-org/plugin-TrackingSpamPrevention/pull/159)

## Checklist

- [NA] I have understood, reviewed, and tested all AI outputs before use
- [NA] All AI instructions respect security, IP, and privacy rules

### Review

- [ ] [Functional review done](https://developer.matomo.org/guides/pull-request-reviews#functional-review-done)
- [ ] [Potential edge cases thought about](https://developer.matomo.org/guides/pull-request-reviews#potential-edge-cases-thought-about) (behaviour of the code with strange input, with strange internal state or possible interactions with other Matomo subsystems)
- [ ] [Usability review done](https://developer.matomo.org/guides/pull-request-reviews#usability-review-done) (is anything maybe unclear or think about anything that would cause people to reach out to support)
- [ ] [Security review done](https://developer.matomo.org/guides/security-in-piwik#checklist)
- [ ] [Wording review done](https://developer.matomo.org/guides/pull-request-reviews#translations-wording-review-done)
- [ ] [Code review done](https://developer.matomo.org/guides/pull-request-reviews#code-review-done)
- [ ] [Tests were added if useful/possible](https://developer.matomo.org/guides/pull-request-reviews#tests-were-added-if-usefulpossible)
- [ ] [Reviewed for breaking changes](https://developer.matomo.org/guides/pull-request-reviews#reviewed-for-breaking-changes)
- [ ] [Developer changelog updated if needed](https://developer.matomo.org/guides/pull-request-reviews#developer-changelog-updated-if-needed)
- [ ] [New documentation added or existing documentation updated](https://developer.matomo.org/guides/pull-request-reviews#documentation-added-if-needed)

sgiehl commented on plugins/Live/vue/src/LivePage/LivePage.vue:
```diff
@@ -11,10 +11,47 @@
       :is="!isWidgetized ? 'ContentBlock' : 'Passthrough'"
       :content-title="!isWidgetized ? translate('Live_VisitorsInRealTime') : undefined"
     >
+<<<<<<< HEAD
```
unresolved merge conflict

sgiehl commented on plugins/CoreHome/templates/_htmlEmailHeader.twig:
```diff
@@ -10,10 +10,10 @@
 <a name="reportTop"></a>
 
-<table style="width:100%; background-color: {{ themeStyles.colorHeaderBackground|e('html_attr') }}; color: {{ themeStyles.colorHeaderText|e('html_attr') }}; padding:10px 0; margin: 0 0 25px 0; height:64px;">
+<table style="width:100%; background-color: {{ themeStyles.getPropertyValue('colorHeaderBackground')|e('html_attr') }}; color: {{ themeStyles.getPropertyValue('colorHeaderText')|e('html_attr') }}; padding:10px 0; margin: 0 0 25px 0; height:64px;">
```
What happens if any plugin template would still use e.g. `themeStyles.colorHeaderBackground` directly?

tzi replied:
If you call `themeStyles.colorHeaderBackground` directly, you have a risk to get an array.

sgiehl replied:
I have checked out all Marketplace plugins & theme to check if that could cause regressions. It seems that no other public plugin/theme uses direct access to themeStyles variables, so this change should imho be okish - even though it introduces a BC issue.

tzi replied:
Thanks for the investigation!

tzi commented on core/Plugin/ThemeStyles.php:
```diff
@@ -41,170 +50,225 @@ class ThemeStyles
+    /**
+     * @var string|array<string>
+     */
+    public $colorBoxShadow = ['rgba(0, 0, 0, 0.1)', 'rgba(0, 0, 0, 0.1)'];
```
ℹ️ If we do a light box-shadow in dark mode, it doesn't look the same at all. More an highlight than stacking shadow.

tzi commented on core/View/HtmlReportEmailHeaderView.php:
```diff
@@ -62,15 +62,15 @@ public function __construct($reportTitle, $prettyDate, $description, $reportMeta
         $view->emailStyles = $emailStyles;
 
-        $view->fontStyle = 'color:' . $themeStyles->colorText . ';font-family:' . $themeStyles->fontFamilyBase . ';';
+        $view->fontStyle = 'color:' . $themeStyles->getPropertyValue('colorText') . ';font-family:' . $themeStyles->getPropertyValue('fontFamilyBase') . ';';
```
ℹ️ We avoid direct access to a color, because it could be defined as an array.

sgiehl replied:
Just as suggestion. We could also change the properties to `protected` and add a magic `__get` method and proxy it to `getPropertyValue`. That way direct access would still work, but would serve the correct result.

chippison replied:
I've added a ticket for [this](https://innocraft.atlassian.net/browse/DEV-20113)

tzi commented on plugins/CoreAdminHome/stylesheets/whatIsNew.less:
```diff
@@ -10,3 +10,22 @@
     width: auto;
   }
 }
+
+.whatisnew .card {
+  .inDarkMode({
+    background: @theme-color-background-tinyContrast;
+  });
+}
+
+.whatisnew-changelist {
+  overflow-y: scroll;
+  max-height: 500px;
+  margin-bottom: 10px;
+  padding-right: 5px;
+  padding-left: 5px;
+}
+
+.whatisnew-btn {
+  float:right;
+  margin-left: 10px;
+}
```
ℹ️ Moved from layout.less

tzi commented on plugins/CoreHome/stylesheets/dataTable/_dataTable.less:
```diff
@@ -414,6 +414,10 @@ table.dataTable td #urlLink {
 }
 
+.dataTableFooterIcons img {
+  filter: @theme-filter-on-illustration;
```
ℹ️ This filter is a trick to invert color of illustration/icons on dark mode. It's valued to `none` on light mode and a color treatment in dark mode.
⚠️ We should not use this on too big elements, because it creates a new stacking context so it could have an impact on the layout (breaking the `z-index` behaviour of its children).

tzi commented on plugins/CoreHome/stylesheets/dataTable/_tableConfiguration.less:
```diff
@@ -15,7 +15,7 @@ a.dropdownConfigureIcon.highlighted {
 }
 
 .tableConfiguration div.configItem:hover {
-  background-color: @color-silver-l95;
```
The `@color-` should be avoided to allow the automatic dark mode.

tzi commented on plugins/CoreHome/vue/src/Alert/alert.less:
```diff
@@ -48,34 +57,50 @@
   background-color: #ebf2eb;
   border-color: #378039;
-  &:before {
+  &::before {
```
ℹ️ fix the old pseudo-element syntax.

sgiehl commented on plugins/Feedback/tests/UI/expected-screenshots/FeedbackQuestion_feedback_failed.png:
Is this a regression? Looks like the "highlight"/"focus" border color changed. That color need to be different for dark mode, but why did the default change?

tzi replied:
This change is a part of a normalization of the colors, I changed some hard-coded colors to have them easily themable.
In this specific case, there was no "focus" color before, I added one in all button, input, optgroup, select, textarea.
At the opposite, the red border is here to notice that there was an error, and it this color is still here if the textarea is not focus. If you click to sned feedback without text, the textarea will be unfocus.

chippison commented on plugins/CoreHome/vue/src/Alert/alert.less:
```diff
@@ -19,6 +19,10 @@
     padding: 15px;
   }
 
+  .inDarkMode({
```
Why are we not using variables in this file?

tzi replied:
Good question!
It's because big green/blue/red zones are not really suitable in dark mode, so it's not the same design.

| Light | Dark |
|---|---|
| <img width="945" height="411" alt="Screenshot 2026-03-30 at 09 25 45" src="https://github.com/user-attachments/assets/91a5a061-e1fa-41ee-99a7-5854d84616f0" /> | <img width="945" height="411" alt="Screenshot 2026-03-30 at 09 26 05" src="https://github.com/user-attachments/assets/019e91fa-e010-48d4-944e-69ba80137ef7" /> |

chippison commented on plugins/CorePluginsAdmin/tests/UI/expected-screenshots/PluginsAdmin_plugins.png:
The red colour on `Uninstall` is no longer applied

tzi replied:
Good catch!
I’ll fix that

chippison commented on plugins/CustomDimensions/tests/UI/expected-screenshots/CustomDimensions_manage_configure_button_disabled.png:
Does the colour of the text change when we disable a button?

tzi replied:
Yes, I updated the background color of the disabled button to improve the contrast.

| before | after |
|---|---|
| <img width="494" height="173" alt="Image" src="https://github.com/user-attachments/assets/c69e4516-951f-4615-8b98-4c747a030a94" /> | <img width="494" height="173" alt="Image" src="https://github.com/user-attachments/assets/807129c9-d708-4297-a565-b77b007bbc88" /> |

But this button background is not consistent with the other. I'll update it.

sgiehl commented on core/Plugin/ThemeStyles.php:
```diff
@@ -41,170 +50,225 @@ class ThemeStyles
     /**
      * @var string
      */
+    public $themeMode = self::AUTO_MODE;
```
Any reason for having this one `public`? There's a get method, and it is set in contructor. If a change would be needed in between, we should introduce a set method...

tzi replied:
Good question.
I think it should be protected, at least.
Do we need a set method? I do not see any reason to change the theme mode between the instantiation of the Theme and the template rendering. Do you want to me to create one just in case?

```suggestion
    protected $themeMode = self::AUTO_MODE;
```

sgiehl replied:
No, not needed for now. We can create it in case we will need it.

sgiehl commented on plugins/CoreAdminHome/tests/UI/expected-screenshots/TrackingCodeGenerator_advanced.png:
The shadow/border of the copy button changed.

tzi replied:
That button’s own CSS in `plugins/CoreHome/stylesheets/coreHome.less:309` never sets border, so it used to rely on the browser/default shared styles. The actual change came from shared Morpheus styling, since I added default/common style on all button, input, optgroup, select, textarea.
Even if we want to change it after, I think it's god think to not depend on browser styles, and we need to define one to be able to switch it on dark mode.

sgiehl commented on plugins/CoreHome/tests/UI/expected-screenshots/SingleMetricView_formatted_metric.png:
The border color of the evolution arrow if gone. That imho makes it harder to view.

tzi replied:
You’re right. I can add it back

tzi replied:
✅ Done!

sgiehl commented on plugins/UserCountryMap/tests/UI/expected-screenshots/VisitorMap_regions.png:
The change in the border color of the select boxes in on purpose?

tzi replied:
That select never sets border, so it used to rely on the browser/default shared styles. The actual change came from shared Morpheus styling, since I added default/common style on all button, input, optgroup, select, textarea.
Even if we want to change it after, I think it's god think to not depend on browser styles, and we need to define one to be able to switch it on dark mode.

tzi replied:
✅ Fixed!

sgiehl commented on plugins/UserId/tests/UI/expected-screenshots/UserId_visitor_profile_popup.png:
Why did the link colors in visitor profile change here?

tzi replied:
You’re right. It’s a regression.
✅ I pushed a fix

sgiehl commented on plugins/TwoFactorAuth/tests/UI/expected-screenshots/TwoFactorAuthUsersManager_edit_with_2fa_reset_confirmed.png:
Isn't the text color of the disabled input now a bit too light? It's a lot hard to read and feels less accessible.

tzi replied:
✅ Fixed!

sgiehl commented on plugins/SitesManager/tests/UI/expected-screenshots/SitesManager_global_url_param_exclusion_add_common.png:
Is the blue border on the button an intentionell focus line?

tzi replied:
Yes. This change is a part of a normalization of the colors, there was no "focus" color defined before, I added one in all button, input, optgroup, select, textarea.

sgiehl commented on plugins/SegmentEditor/tests/UI/expected-screenshots/SegmentSelectorEditorTest_saved_details.png:
I guess the change of the background color of the segment editor is intentionell?

tzi replied:
Yes, there isn’t really a dedicated warm beige token in the theme today. So I used one of the available background tokens and they are mostly neutral/cool.

sgiehl commented on plugins/SegmentEditor/tests/UI/expected-screenshots/SegmentSelectorEditorTest_enabled_create_realtime_segments_saved.png:
The background of the search input makes it imho harder to read

tzi replied:
Good catch. I’ll fix it

tzi replied:
✅ Fixed

sgiehl commented on plugins/UsersManager/tests/UI/expected-screenshots/UsersManager_delete_single_confirm.png:
Might be unrelated, but I'm wondering why the text changed - or why the text change didn't cause a failure before 🤔

tzi replied:
I’ll improve it 👍

tzi replied:
✅ Fixed, it now uses the same styles than disabled buttons

sgiehl commented on plugins/UsersManager/tests/UI/expected-screenshots/UsersManager_invite_confirm.png:
Same text change here 🧐

tzi replied:
I’ll improve it 👍

tzi replied:
✅ Fixed too

chippison replied:
@tzi 
I think the text change Stefan was talking about here is the change from `Please enter your password to confirm this change` to `Please re-authenticate to confirm this change`. I'm also curious on why this text change happened

sgiehl replied:
Maybe the screenshot test has a too high error tollerance

sgiehl commented on plugins/UsersManager/tests/UI/expected-screenshots/UsersManager_permissions_all_sites_access.png:
The Bulk actions button/drop-down isn't readable anymore.

tzi replied:
✅ Fixed

sgiehl commented on tests/UI/expected-screenshots/Comparison_multi_row_evolution.png:
The darker border on the input here doesn't make it look much better. But guess that's more an issue of spacing that makes it look bad.

tzi replied:
Yes, it needs spacing 😅 
Seems minor, we can certainly keep this feedback for another PR

sgiehl commented on tests/UI/expected-screenshots/Menus_mobile_left.png:
Is the menu highlight missing her intentional?

tzi replied:
I didn’t find where this regression come from, but I added a background on the active collapsible-header.

tzi replied:
✅ Fixed!

sgiehl commented on tests/UI/expected-screenshots/Menus_mobile_left_admin.png:
Main menu items here also lost the background.

tzi replied:
✅ Fixed!

sgiehl commented on tests/UI/expected-screenshots/PeriodSelector_custom_comparison_url.png:
The blue border here looks a bit misplaced - even if it's a focus border 🤔

tzi replied:
I don't think so. There could be more spaces with the arrows, is it what you mean? 🤔

sgiehl commented on tests/UI/expected-screenshots/Theme_home.png:
Did the font here change on purpose or why does the screenshot differ that much?

tzi replied:
✅ Fixed (it came from a font change in the ExampleTheme)

sgiehl commented on plugins/UsersManager/tests/UI/expected-screenshots/UserSettings_signup_success.png:
The description text for the Theme option needs to be added to the javascript translations.

tzi replied:
✅ Fixed!

sgiehl commented on plugins/UsersManager/tests/UI/expected-screenshots/UserSettings_load_security_new_token.png:
The text color of the disabled button feels inconsistent. Sometimes it's white - sometimes grey

tzi replied:
Found, it the white color came from the `<input type="button />`, I’ll fix it.

tzi replied:
✅ Fixed!

sgiehl commented on plugins/Morpheus/images/chart_line_edit_dark.svg:
Any reason for not moving this to the font? That way the color could be changed easily instead of having two icons 🤔

tzi replied:
I hadn't thought of that solution, but icon fonts are sort of an anti-pattern so it's not the first thing that comes to mind.

sgiehl commented on plugins/Live/tests/UI/expected-screenshots/Live_visitor_profile_action_details.png:
I think I already mentioned the change link color in the profile, but just to be sure I did,... 🤔

tzi replied:
✅ It’s fixed

sgiehl reviewed (changes requested):
I had a look through all the changed screenshots. I've tried to highlight all changes that did not appear to be obvious to me. Feel free to simply ignore/resolve those that were done on purpose and in consultation with the UX team. Others really look like regressions.

I did not really had a brief look through the UI yet, but there also seems to be some parts, that are hard to read in dark mode. But I guess we can also improve those over time later. 
One example:
<img width="1124" height="374" alt="Image" src="https://github.com/user-attachments/assets/ed159f5c-658f-4015-8fc6-8a15b1b1bb27" />

chippison commented on plugins/Marketplace/tests/UI/expected-screenshots/Marketplace_paid_plugin2_plugin_details_superuser.png:
This look like it has an overlay or has the font colours changed ihe following component?

tzi replied:
- before: the modal used hard-coded black text
  - now: the modal uses themed text colors for dark-mode support
  
I had to change it to be able to have the dark mode and there is no reason to not use the default text color on white background.

chippison commented on plugins/MultiSites/tests/UI/expected-screenshots/AllWebsitesDashboard_dashboard.png:
Do we add the border for the select box beside `Evolution` column?

tzi replied:
✅ Fixed!

chippison commented on plugins/PrivacyManager/tests/UI/expected-screenshots/PrivacyManager_gdpr_tools_visits_showprofile.png:
The links have also changed colours here,
Just added this here because this does not seem to be a global issue

tzi replied:
✅ It's fixed, the links now use the generic link color (there was black at the time of this message)

chippison commented on plugins/SegmentEditor/tests/UI/expected-screenshots/SegmentSelectorEditorTest_1_selector_open.png:
Is this intentional? It looks to me that the search bar is disabled from initial view

tzi replied:
✅ Fixed

chippison commented on plugins/TwoFactorAuth/tests/UI/expected-screenshots/TwoFactorAuth_twofa_forced_step3.png:
Should the text in the Confirm button change colour as well? Just as in step1.png above?

tzi replied:
✅ Good catch. it's fixed!

chippison commented on plugins/TwoFactorAuth/tests/UI/expected-screenshots/TwoFactorAuth_twofa_setup_step3.png:
Same here for the text colour on Confirm btn

tzi replied:
✅ Fixed!

chippison commented on plugins/UsersManager/tests/UI/expected-screenshots/UserSettings_already_signed_up.png:
The translation string for the `UsersManager_ThemeModeHelp` was not loaded in JS

tzi replied:
✅ Fixed!

chippison commented on plugins/UsersManager/tests/UI/expected-screenshots/UserSettings_load_security.png:
Shouldn't 'Save' button have a different text colour?

tzi replied:
✅ Fixed

chippison commented on tests/UI/expected-screenshots/UIIntegrationTest_admin_settings_general.png:
Should there be a blue border on the text input for IPs we want to block (In Login section)

tzi replied:
✅ Fixed!

chippison reviewed (commented):
I have also done some functional tests:
 - Going through most pages in dark mode
   - admin
   - tagmanager
   - dashboard
 - Creating users with different mode settings 
   - switching between users was working correctly
 - Testing persistence of setting by switching browsers was also working correctly

tzi committed:
Add dark mode theme and UI styling

tzi committed:
Add theme preference support in UsersManager

tzi committed:
Update UI screenshots for dark mode

chippison committed:
fixing issue where getPropertyValue can return null or throw an error
 - also making sure toLessCode uses the same rule to resolve property value instead of just always assuming the array has two values
 - fixing the issue with scrollbars

chippison committed:
fixing issue with visitor map dropdown not showing properly in dark mode

sgiehl commented on plugins/SegmentEditor/stylesheets/segmentation.less:
```diff
@@ -443,12 +444,13 @@ div.scrollable {
+    background: @theme-color-background-contrast;
+    border: 1px solid @theme-color-border;
+    font-size: 11px;
+    color: #999;
```
Shouldn't that also use a variable?

chippison replied:
This might be intended as that font color looks good/readable in both light and dark mode

tzi replied:
Yes #999 is kind of the middle, so it was ok in dark and light mode 😅

sgiehl commented on tests/UI/expected-screenshots/Menus_admin_loaded.png:
This still looks like a regression. The main menu item is now also highlighted. This seems only to be the case in the admin menu. In the reporting menu this is not the case, which also feels inconsistent.

chippison replied:
This seems to happen to the reporting menu as well for me. but Im just getting confirmation from Pedro and Valeriia. I'll update later when I get confirmation

chippison replied:
Just got confirmation from PO that this is all good in dark mode. There are more changes that will happen for the mobile menu later on and those will be handled in a separate ticket

sgiehl reviewed (dismissed):
I had another look through the screenshot and left some last comments. The possible regression in the menu is something we should clarify and fix if needed. Everything else can also be fixed later.

chippison committed:
Add 'Match browser' option to settings page (#24339)

* adding 'Match browser' option to Settings for dark/light mode

* adding tests and screenshots

* simplifying themes.js and tests

chippison committed:
Dark mode auto refresh visuals (#24350)

* adding reactive visualizations

* fixing issue with sparklines in rowevolution graph do not change the colours when preferences change

* fixing issues in transitions.js and dataTable_rowactions.js

* fixing issue found from feedback
